Docs: Shopify web pixel (#8004)
PostHog/posthog.com: contents/docs/libraries/shopify.mdx

---
title: Shopify
icon: ../../images/docs/integrate/frameworks/shopify.svg
---

import eventLight from '../../images/tutorials/shopify/event-light.png';
import eventDark from '../../images/tutorials/shopify/event-dark.png';

> Tracking your [Shopify](https://www.shopify.com/) store with PostHog can help you improve the user experience and increase conversion rates.

Reviewer: No need for this
```suggestion
```

## Setting up a custom web pixel

The simplest way to add PostHog to Shopify is using a [custom web pixel](https://shopify.dev/docs/api/web-pixels-api#custom-web-pixels). To set this up, first, go to the [customer events](https://admin.shopify.com/settings/customer_events) tab in your store settings, click "Add custom pixel," and give your pixel a name.

Next, add code that contains:

1. The customer events you want to subscribe to. You can find a complete list in Shopify's [Web Pixels API documentation](https://shopify.dev/docs/api/web-pixels-api/standard-events).

2. Your PostHog JavaScript snippet which you can get from [your project settings](https://us.posthog.com/settings/project#snippet).

3. The rest of your PostHog capture logic. 

> For example, we use a custom pixel in [our merch store](https://merch.posthog.com/) to identify customers and capture a `merch store order submitted` event with details like `userID`, `order_number`, `product_id`, `price`, and more.

Reviewer: ```suggestion
As an example, here is the code for subscribing to a `checkout_completed` event. Every time a user checks out, we capture an event. We also call [`posthog.identify()`](/docs/product-analytics/identify) so that we can track the user across different sessions:
```

  posthog.init('<ph_project_api_key>',{api_host:'<ph_instance_address>'})
  const checkout = event.data.checkout;

  posthog.identify(checkout.email, {
      email: checkout.email,
      first_name: checkout.billingAddress.firstName,
      last_name:  checkout.billingAddress.lastName,
    }
  )
  posthog.capture('merch store order submitted', {
    value: checkout.totalPrice.amount,
    created_at: event.timestamp,
    order_number: checkout.order.id,
    userId: event.clientId,
    order_id: checkout.order.id,
    products: checkout.lineItems.map((item) => {
      return {
        product_id: item.id,
        title: item.variant.product.title,
        quantity: item.quantity,
        price: item.variant.price.amount,
        currency: item.variant.price.currencyCode,
        sku: item.variant.sku,
        size: item.variant.product.size
      };
    }),
  })
});
```

> With this, every time a Shopify `checkout_completed` event fires (when a user completes a purchase), PostHog captures the event.

Reviewer: ```suggestion
In the above code, we capture the event with details like `userID`, `order_number`, `product_id`, `price`, and more. This is then viewable in PostHog:
```

<ProductScreenshot 
  imageLight={eventLight} 
  imageDark={eventDark} 
  alt="Shopify events in PostHog" 
/>

## Installing PostHog in a Shopify theme

To have unrestricted access to the features of PostHog in your Shopify store, you can install PostHog in your Shopify theme. To do this:

1. Get your PostHog JavaScript snippet from [your project settings](https://us.posthog.com/settings/project#snippet)
1. Login to your Shopify dashboard
1. Go to 'Online Store' -> 'Themes' (see image below)
1. On your theme, click 'Actions' -> 'Edit code' (see image below)

![Shopify Dashboard](../../images/tutorials/shopify/shopify-dashboard.png)

1. You should now be seeing a code editor. Click on `theme.liquid` under 'Layout' on the left sidebar (see image below)
1. Navigate until you see the closing `</head>` tag. Paste your snippet there, before that tag, like in the image below:

![Shopify Dashboard](../../images/tutorials/shopify/snippet.png)

1. Click the green save button on the top right and you're good to go - PostHog should now capturing events on your Shopify store!

> > To confirm PostHog is configured correctly, visit your store and then check if the events from your session appear in PostHog. This may take a few minutes.

Reviewer: ```suggestion
> To confirm PostHog is configured correctly, visit your store and then check if the events from your session appear in the [PostHog activity tab](https://us.posthog.com/events). This may take a few minutes.
```

### Advanced tracking

> It can be useful to track additional events on your Shopify store, such as when a user adds an item to their cart or completes a purchase.
> 
> Although [autocapture](/docs/product-analytics/autocapture) does capture these events, you may need to add additional metadata. This information can be used in further analysis in aggregate or when understanding behavior of an individual user.
> 
> > The following requires some knowledge of HTML and Shopify's [Liquid templating engine](https://shopify.dev/docs/api/liquid).

Reviewer: ```suggestion
> This section requires some knowledge of HTML and Shopify's [Liquid templating engine](https://shopify.dev/docs/api/liquid).

It can be useful to track additional events on your Shopify store, such as when a user adds an item to their cart or completes a purchase.

Although [autocapture](/docs/product-analytics/autocapture) does capture these events, you may need to add additional metadata. This information can be used in further analysis in aggregate or when understanding behavior of an individual user.
```
